x/gamm/keeper/pool.go

```diff
@@ -94,7 +95,7 @@ func (k Keeper) newBalancerPool(ctx sdk.Context, balancerPoolParams types.Balanc
 		return nil, sdkerrors.Wrapf(types.ErrPoolAlreadyExist, "pool %d already exist", poolId)
 	}
 
-	err = k.SetPool(ctx, pool)
+	err = k.SetPool(ctx, &pool)
```

Comment:
Is there a reason behind changing to pointer implementation of Pool interface?

Reply:
If not it omits an error saying that the methods of poolI has pointer receiver

Reply:
That makes sense to me, several pool methods are mutative, so they need a pointer receiver.  (And the SetPool interface needs what matches the interface)